lib/middleware/route-command.js

```diff
@@ -6,7 +6,37 @@ const Command = require('../slack/command');
 module.exports = function route(req, res, next) {
   req.log.debug({ command: req.body }, 'Received slash command');
 
-  const command = new Command(req.body);
+  let callback;
+  let command;
+
+  // Skip timeout handling if request is a pending command
+  if (req.query.trigger_id) {
+    callback = null;
+  } else {
+    let timeout;
+
+    callback = (message) => {
+      req.log({ message }, 'Responding to slash command');
+      res.status(200).json(message);
+
+      // Sending response, so clear the request timeout.
+      clearTimeout(timeout);
+
+      // Any further responses should use the delayed response callback
+      command.delay();
+    };
+
+    // Slack has a 3000ms timeout, so if we don't respond within that window, ack
+    // the command and use the delayed response URL.
+    timeout = setTimeout(() => {
+      req.log('Command taking too long. Switching to delayed response.');
+
+      // Ack the command
+      command.respond({ response_type: 'in_channel' });
+    }, 2500);
```

Comment:
I wonder if 2500s may already be too long (because the response has to make its way back to Slack), but i guess we can only try it out and see :)